src/coreclr/jit/lsra.h

```diff
@@ -2064,15 +2066,43 @@ class LinearScan : public LinearScanInterface
     //
     static regMaskTP calleeSaveRegs(RegisterType rt)
     {
-        return varTypeIsIntegralOrI(rt) ? RBM_INT_CALLEE_SAVED : RBM_FLT_CALLEE_SAVED;
+        if (varTypeUsesIntReg(rt))
+        {
+            return RBM_INT_CALLEE_SAVED;
+        }
+#if defined(TARGET_XARCH) && defined(FEATURE_SIMD)
+        else if (varTypeUsesMaskReg(rt))
+        {
+            return RBM_MSK_CALLEE_SAVED;
+        }
+#endif // TARGET_XARCH && FEATURE_SIMD
+        else
+        {
+            assert(varTypeUsesFloatReg(rt));
+            return RBM_FLT_CALLEE_SAVED;
+        }
     }
 
     //------------------------------------------------------------------------
     // callerSaveRegs: Get the set of caller-save registers of the given RegisterType
     //
     regMaskTP callerSaveRegs(RegisterType rt) const
     {
-        return varTypeIsIntegralOrI(rt) ? RBM_INT_CALLEE_TRASH : RBM_FLT_CALLEE_TRASH;
+        if (varTypeUsesIntReg(rt))
```

Comment:
can we have a cache that maps the `rt -> regMaskTP`? Then, this can be changed to `return cachedMap[rt]];`?

Reply:
Have a prototype for that here: https://github.com/dotnet/runtime/pull/89351

Reply:
Have pulled that into this PR as it significantly reduces the TP regression